test/autorest/complexgroup/dictionary_test.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package complexgrouptest
+
+import (
+	"context"
+	"generatortests/autorest/generated/complexgroup"
+	"net/http"
+	"testing"
+)
+
+func getDictionaryOperations(t *testing.T) complexgroup.DictionaryOperations {
+	client, err := complexgroup.NewDefaultClient(nil)
+	if err != nil {
+		t.Fatalf("failed to create complex client: %v", err)
+	}
+	return client.DictionaryOperations()
+}
+
+func TestDictionaryGetEmpty(t *testing.T) {
+	client := getDictionaryOperations(t)
+	result, err := client.GetEmpty(context.Background())
+	if err != nil {
+		t.Fatalf("GetEmpty: %v", err)
+	}
+	val := complexgroup.DictionaryWrapper{DefaultProgram: map[string]*string{}}
+	expected := &complexgroup.DictionaryGetEmptyResponse{
+		StatusCode:        http.StatusOK,
+		DictionaryWrapper: &val,
+	}
+	deepEqualOrFatal(t, result, expected)
+}
+
+func TestDictionaryGetNotProvided(t *testing.T) {
+	client := getDictionaryOperations(t)
+	result, err := client.GetNotProvided(context.Background())
+	if err != nil {
+		t.Fatalf("GetNotProvided: %v", err)
+	}
+	val := complexgroup.DictionaryWrapper{}
+	expected := &complexgroup.DictionaryGetNotProvidedResponse{
+		StatusCode:        http.StatusOK,
+		DictionaryWrapper: &val,
+	}
+	deepEqualOrFatal(t, result, expected)
+}
+
+func TestDictionaryGetNull(t *testing.T) {
+	client := getDictionaryOperations(t)
+	result, err := client.GetNull(context.Background())
+	if err != nil {
+		t.Fatalf("GetNull: %v", err)
+	}
+	expected := &complexgroup.DictionaryGetNullResponse{
+		StatusCode:        http.StatusOK,
+		DictionaryWrapper: &complexgroup.DictionaryWrapper{},
```

Comment:
This looks identical to `GetNotProvided()`, is this due to more `omitempty` shenanigans?

Reply:
Actually, I would say this is more of a testserver shenanigan, the endpoint isn't expecting a true null it's expecting a null field in DictionaryWrapper. 

So I looked into what GetNotProvided returns versus GetNull. 
GetNull is expecting: {"defaultProgram":null}
GetNotProvided is expecting: {}

I also just removed `omitempty` locally and nothing changed, both tests still run successfully. I think this could still be related to how we're unmarshalling though.

Reply:
Ah I see, thanks for the explanation.  With this info then I would expect this test to fail since its dictionary is empty.  If you remove the `omitempty` from the `DefaultProgram` field of `DictionaryWrapper` does it work as expected?

Reply:
Disregard the above comment

Reply:
The endpoints are expecting this: 
```
 else if (req.params.scenario === 'empty') {
      coverage['getComplexDictionaryEmpty']++;
      res.status(200).type('json').end('{"defaultProgram":{}}');
    } else if (req.params.scenario === 'null') {
      coverage['getComplexDictionaryNull']++;
      res.status(200).type('json').end('{"defaultProgram":null}');
```

Reply:
So both of them are pretty similar, I think they get unmarshalled into the same format which is why the tests pass, regardless of the `omitempty` tag. Also, I'm going to update the code to remove the `val` var in GetNotProvided since it's unnecessary.

Reply:
It seems similar to the issue we were looking at with arrays yesterday and the whole "empty array vs nil array" and is that meaningful.  For now I think just open an issue to track this and we'll look at it later.

Reply:
Done. Issue is here: #255